packages/autorest.python/autorest/codegen/templates/model_base.py.jinja2

```diff
@@ -296,10 +310,17 @@ def get_deserializer(annotation: typing.Any, rf: typing.Optional["_RestField"] =
 
 
 def _get_model(module_name: str, model_name: str):
-    models = {k: v for k, v in sys.modules[module_name].__dict__.items() if isinstance(v, type)}
+    models = {
+        k: v
+        for k, v in sys.modules[module_name].__dict__.items()
+        if isinstance(v, (type, typing._GenericAlias))  # type: ignore
+    }
     module_end = module_name.rsplit(".", 1)[0]
-    module = sys.modules[module_end]
-    models.update({k: v for k, v in module.__dict__.items() if isinstance(v, type)})
+    models.update({
+        k: v
+        for k, v in sys.modules[module_end].__dict__.items()
+        if isinstance(v, (type, typing._GenericAlias))  # type: ignore
```

Comment:
what is this added code for? I don't know if we should access a private model in the `typing` namespace

Reply:
It is for type alias like:
```
union MyNamedUnion {
  one: Model1,
  two: Model2,
}
```